Enable trunk

det opened the pull request:
- Enable trunk with the following linters:
    - actionlint
    - eslint
    - gitleaks
    - markdownlint
    - prettier

- Disable running prettier via eslint. See [rationale](https://github.com/trunk-io/configs#1-turn-off-formatting-errors-use-autoformatters-instead).
- Trunk ignored some gitleaks checks that are intentional leaks.
- Change the lint workflow to run trunk instead of eslint.

jasonLaster commented on .trunk/trunk.yaml:
```diff
@@ -0,0 +1,22 @@
+    - eslint@8.9.0
+    - gitleaks@8.3.0
+    - markdownlint@0.31.1
+    - prettier@2.5.1
```
Should this stay in sync with `package.json`?

det replied:
Yeah, ideally they should stay in sync if you run any of these outside of trunk.

jasonLaster commented on .eslintrc.json:
```diff
@@ -4,11 +4,7 @@
-  "extends": [
-    "plugin:react/recommended",
-    "plugin:react-hooks/recommended",
-    "plugin:prettier/recommended"
```
I assume this is now in the formatter?

det replied:
Correct, see https://github.com/trunk-io/configs#1-turn-off-formatting-errors-use-autoformatters-instead

jasonLaster reviewed (commented):
Looking good. 

Could you add a section in the README.md for running the linter and formatter locally?

abstractalgo commented:
Hi @det, as a FYI, the Vercel build is failing due to lint errors on files that were in `.eslintignore` previously. We actually really need some files untouched by linters/formatters.

Thanks for the effort so far 🙌

jasonLaster commented:
Looking through the errors, this is what I see:

```




./src/devtools/client/debugger/src/workers/parser/tests/fixtures/component.js
18:21:43.504 | 19:3  Error: Your render method should have a return statement  react/require-render-return
18:21:43.504 | 63:34  Error: Expected { after 'if' condition.  curly
18:21:43.504 |  
18:21:43.504 | ./src/devtools/client/debugger/src/workers/parser/tests/fixtures/frameworks/reduxLibrary.js
18:21:43.504 | 39:3  Error: Parsing error: '}' expected.
18:21:43.504 |  
18:21:43.505 | ./src/devtools/client/debugger/src/workers/parser/tests/fixtures/jsx.js
18:21:43.505 | 4:4  Error: 'Item' is not defined.  react/jsx-no-undef
18:21:43.505 |  
18:21:43.505 | ./src/devtools/client/debugger/src/workers/parser/tests/fixtures/scopes/class-property.js
18:21:43.505 | 6:10  Error: Parsing error: Expression expected.
18:21:43.505 |  
18:21:43.505 | ./src/devtools/client/debugger/src/workers/parser/tests/fixtures/scopes/flowtype-bindings.js
18:21:43.505 | 7:38  Error: Parsing error: '=>' expected.
18:21:43.505



./src/ui/utils/sourcemapVisualizer.js
--
18:21:43.515 | 14:45  Error: Expected { after 'for' condition.  curly
18:21:43.515 | 15:45  Error: Expected { after 'for' condition.  curly
18:21:43.515 | 44:35  Error: Expected { after 'if' condition.  curly
18:21:43.515 | 46:31  Error: Expected { after 'if' condition.  curly
18:21:43.515 | 48:29  Error: Expected { after 'if' condition.  curly
18:21:43.515 | 56:33  Error: Expected { after 'if' condition.  curly
```

det commented:
- Reverted moving the ignroes to trunk.yaml, it seems the Vercel workflow runs eslint outside of trunk so it will not pick up the ignores from there.
- Added information to README.md on installing and running trunk. We are also taking your advice on adding the trunk launcher to npm, so the installing trunk launcher step will be unnecessary in the next week.

abstractalgo commented:
> * Reverted moving the ignroes to trunk.yaml, it seems the Vercel workflow runs eslint outside of trunk so it will not pick up the ignores from there.
> * Added information to README.md on installing and running trunk. We are also taking your advice on adding the trunk launcher to npm, so the installing trunk launcher step will be unnecessary in the next week.

There's a way to ignore eslint on builds (https://nextjs.org/docs/api-reference/next.config.js/ignoring-eslint), we can try that, or a custom build command on Vercel, if required.

det commented:
Trunk works fine with the ignores inside of .prettierignore and .eslintignore, it's just a bit nicer to have them in trunk.yaml IMO. I think this should be good now to go unless you see any other changes to be made. The ignores can be tweaked in the future if desired.
